src/orion/algo/ax/axoptimizer.py

```diff
@@ -84,12 +107,17 @@ def suggest(self, num):
         """
         # TODO: Adapt this to your algo
         trials = []
-        while len(trials) < num and not self.is_done:
-            seed = tuple(self.rng.randint(0, 1000000, size=3))
-            new_trial = self.format_trial(self.space.sample(1, seed=seed)[0])
-            if not self.has_suggested(new_trial):
-                self.register(new_trial)
-                trials.append(new_trial)
+        with self.get_client() as client:
+            while len(trials) < num and not self.is_done:
+                parameters, complete = client.get_next_trial()
+                if complete == True:
+                    break
```

Comment:
Mélangé `get_next_trials()` et `get_next_trial()`. `client.get_current_trial_generation_limit()` devrait être appelé autre part, peut-être dans `self.is_done`
```suggestion
                parameters, _ = client.get_next_trial()
```